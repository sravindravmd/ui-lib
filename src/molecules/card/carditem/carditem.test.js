import React from 'react'
import { render, cleanup } from '@testing-library/react'
import CardItem, { DetailItem, GeneralItem, CategoryItem } from './index'

afterEach(cleanup)

describe('<CardItem/>', () => {
  const imgUrl =
    'https://images.unsplash.com/photo-1603881568692-1429786f5ce1?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60'
  it('renders Detail Card Item correctly', () => {
    const { container } = render(
      <DetailItem title='name' property='property' />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders General Card Item without action correctly', () => {
    const { container } = render(
      <GeneralItem
        title='title'
        subtitle='subtitle'
        imgUrl={imgUrl}
        imgAltText='card-item'
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders General Card Item with action correctly', () => {
    const { container } = render(
      <GeneralItem
        title='title'
        subtitle='subtitle'
        imgUrl={imgUrl}
        imgAltText='card-item'
        isActionItem={true}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders Category Card Item correctly', () => {
    const { container } = render(
      <CategoryItem text='Subcategories' value='51' />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders Card Item with valid variant correctly', () => {
    const { container } = render(
      <CardItem variant='detailItem' title='name2' property='property2' />
    )
    expect(container).toMatchSnapshot()
  })
})
