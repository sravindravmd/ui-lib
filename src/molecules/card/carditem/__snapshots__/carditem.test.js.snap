// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardItem/> renders Card Item with valid variant correctly 1`] = `
<div>
  <div
    class="sc-hKgILt fliuLD"
  >
    <div
      class="sc-gsTCUz zWUzB"
    >
      name2
    </div>
    <div
      class="sc-dlfnbm kwrUSG"
    >
      property2
      <span
        class="sc-bdfBwQ gSwybW"
      >
        <svg
          fill="none"
          height="12"
          viewBox="0 0 12 12"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.1207 0.5C10.322 0.5 10.5019 0.534424 10.6651 0.599143C10.8318 0.665267 10.9775 0.758527 11.1055 0.880373C11.2306 0.99953 11.3254 1.13664 11.3924 1.29517C11.4584 1.4513 11.4962 1.62903 11.5 1.83399C11.4997 1.9981 11.482 2.13089 11.4534 2.23703C11.4211 2.35663 11.3767 2.46508 11.3212 2.5639C11.263 2.66746 11.1903 2.77097 11.1014 2.87419C11.0957 2.87357 11.0864 2.87057 11.0774 2.86154L9.11944 0.906298C9.11633 0.903193 9.11484 0.900551 9.11397 0.898418C9.11298 0.895975 9.11259 0.893595 9.1126 0.891709C9.11261 0.891298 9.11262 0.890956 9.11265 0.890679C9.20423 0.805589 9.30027 0.735453 9.40085 0.678821C9.50346 0.621047 9.61006 0.576522 9.72129 0.54471C9.81334 0.518385 9.94349 0.5 10.1207 0.5ZM7.08433 2.92294C7.27931 2.72559 7.59767 2.72471 7.79374 2.92097L9.07302 4.2015C9.26819 4.39686 9.26803 4.71345 9.07267 4.90861L3.46161 10.514L1.61597 11.0295C1.23639 11.1355 0.889078 10.7812 1.00268 10.4038L1.5765 8.49766L7.08433 2.92294Z"
            fill="#121213"
            stroke="#121213"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`<CardItem/> renders Category Card Item correctly 1`] = `
<div>
  <div
    class="sc-jrAGrp bsjwUw"
  >
    <div
      class="sc-kEjbxe bLSkeB"
    >
      Subcategories
    </div>
    <div
      class="sc-iqHYGH jpZlsZ"
    >
      51
    </div>
  </div>
</div>
`;

exports[`<CardItem/> renders Detail Card Item correctly 1`] = `
<div>
  <div
    class="sc-hKgILt fliuLD"
  >
    <div
      class="sc-gsTCUz zWUzB"
    >
      name
    </div>
    <div
      class="sc-dlfnbm kwrUSG"
    >
      property
      <span
        class="sc-bdfBwQ gSwybW"
      >
        <svg
          fill="none"
          height="12"
          viewBox="0 0 12 12"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.1207 0.5C10.322 0.5 10.5019 0.534424 10.6651 0.599143C10.8318 0.665267 10.9775 0.758527 11.1055 0.880373C11.2306 0.99953 11.3254 1.13664 11.3924 1.29517C11.4584 1.4513 11.4962 1.62903 11.5 1.83399C11.4997 1.9981 11.482 2.13089 11.4534 2.23703C11.4211 2.35663 11.3767 2.46508 11.3212 2.5639C11.263 2.66746 11.1903 2.77097 11.1014 2.87419C11.0957 2.87357 11.0864 2.87057 11.0774 2.86154L9.11944 0.906298C9.11633 0.903193 9.11484 0.900551 9.11397 0.898418C9.11298 0.895975 9.11259 0.893595 9.1126 0.891709C9.11261 0.891298 9.11262 0.890956 9.11265 0.890679C9.20423 0.805589 9.30027 0.735453 9.40085 0.678821C9.50346 0.621047 9.61006 0.576522 9.72129 0.54471C9.81334 0.518385 9.94349 0.5 10.1207 0.5ZM7.08433 2.92294C7.27931 2.72559 7.59767 2.72471 7.79374 2.92097L9.07302 4.2015C9.26819 4.39686 9.26803 4.71345 9.07267 4.90861L3.46161 10.514L1.61597 11.0295C1.23639 11.1355 0.889078 10.7812 1.00268 10.4038L1.5765 8.49766L7.08433 2.92294Z"
            fill="#121213"
            stroke="#121213"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`<CardItem/> renders General Card Item with action correctly 1`] = `
<div>
  <div
    class="sc-eCssSg fVAgpT"
  >
    <div
      class="sc-jSgupP fQKavk"
    >
      <img
        alt="card-item"
        src="https://images.unsplash.com/photo-1603881568692-1429786f5ce1?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
      />
    </div>
    <div
      class="sc-gKsewC jmUenl"
    >
      <div
        class="sc-iBPRYJ eHHpwo"
      >
        title
      </div>
      <div
        class="sc-fubCfw dXENNU"
      >
        subtitle
      </div>
    </div>
    <div
      class="sc-pFZIQ JNXgK"
    >
      <svg
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.74897 13.1903C4.45607 13.4831 4.45607 13.958 4.74897 14.2509C5.04186 14.5438 5.51673 14.5438 5.80963 14.2509L4.74897 13.1903ZM11.3096 8.75095C11.6025 8.45805 11.6025 7.98318 11.3096 7.69029C11.0167 7.39739 10.5418 7.39739 10.2489 7.69029L11.3096 8.75095ZM5.80963 14.2509L11.3096 8.75095L10.2489 7.69029L4.74897 13.1903L5.80963 14.2509Z"
          fill="#121213"
        />
        <path
          d="M5.86835 2.24897C5.57546 1.95608 5.10058 1.95607 4.80769 2.24896C4.51479 2.54185 4.51479 3.01672 4.80768 3.30962L5.86835 2.24897ZM10.2492 8.75126C10.5421 9.04416 11.017 9.04416 11.3099 8.75127C11.6027 8.45838 11.6028 7.98351 11.3099 7.69061L10.2492 8.75126ZM4.80768 3.30962L10.2492 8.75126L11.3099 7.69061L5.86835 2.24897L4.80768 3.30962Z"
          fill="#121213"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`<CardItem/> renders General Card Item without action correctly 1`] = `
<div>
  <div
    class="sc-eCssSg iKlbQh"
  >
    <div
      class="sc-jSgupP ksflJx"
    >
      <img
        alt="card-item"
        src="https://images.unsplash.com/photo-1603881568692-1429786f5ce1?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
      />
    </div>
    <div
      class="sc-gKsewC gaeMlE"
    >
      <div
        class="sc-iBPRYJ eHHpwo"
      >
        title
      </div>
      <div
        class="sc-fubCfw dXENNU"
      >
        subtitle
      </div>
    </div>
  </div>
</div>
`;
